modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,341 @@
+
+		if ( count( $attachment_ids ) > 1 ) {
+			/*
+			 * Warm the object cache with post and meta information for all found.
+			 * images to avoid making individual database calls.
+			 */
+			_prime_post_caches( $attachment_ids, false, true );
+		}
+
+		// Iterate through the matches in order of occurrence as it is relevant for whether or not to lazy-load.
+		foreach ( $matches as $match ) {
+			// Filter an image match.
+			if ( isset( $images[ $match[0] ] ) ) {
+				$filtered_image = $match[0];
+				$attachment_id  = $images[ $match[0] ];
+				/**
+				 * Filters img tag that will be injected into the content.
+				 *
+				 * @param string $filtered_image the img tag with attributes being created that will
+				 *                                    replace the source img tag in the content.
+				 * @param string $context Optional. Additional context to pass to the filters.
+				 *                        Defaults to `current_filter()` when not set.
+				 * @param int $attachment_id the ID of the image attachment.
+				 *
+				 * @since 1.0.0
+				 */
+				$filtered_image = apply_filters( 'dominant_color_img_tag_add_adjust', $filtered_image, $context, $attachment_id );
+
+				if ( $filtered_image !== $match[0] ) {
+					$content = str_replace( $match[0], $filtered_image, $content );
+				}
+			}
+		}
+
+		return $content;
+	}
+
+	$filters = array( 'the_content', 'the_excerpt', 'widget_text_content', 'widget_block_content' );
+	foreach ( $filters as $filter ) {
+		add_filter( $filter, 'dominant_color_filter_content_tags', 20 );
+	}
+
+	add_filter( 'dominant_color_img_tag_add_adjust', 'dominant_color_tag_add_adjust', 10, 3 );
+}
+
+/**
+ * Add CSS needed for to show the dominant color as an image background.
+ */
+function dominant_color_add_inline_style() {
+	$handle = 'dominant-color-styles';
+	wp_register_style( $handle, false );
+	wp_enqueue_style( $handle );
+	$custom_css = 'img[data-dominantcolor]:not(.has-transparency) { background-color: var(--dominant-color); background-clip: content-box, padding-box; }';
+	wp_add_inline_style( $handle, $custom_css );
+}
+add_filter( 'wp_enqueue_scripts', 'dominant_color_add_inline_style' );
+
+
+/**
+ * Overloads wp_image_editors() to load the extended classes.
+ *
+ * @return string[]
+ */
+function dominant_color_set_image_editors() {
+
+	require_once 'class-dominant-color-image-editor-gd.php';
+	require_once 'class-dominant-color-image-editor-imagick.php';
+
+	return array( 'Dominant_Color_Image_Editor_GD', 'Dominant_Color_Image_Editor_Imagick' );
+}
+
+/**
+ * Get dominant color of image
+ *
+ * @param integer $id the image id.
+ * @param string  $default_color default color.
+ *
+ * @return string
+ */
+function dominant_color_get( $id, $default_color = 'eee' ) {
+
+	add_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	$file = get_attached_file( $id );
+
+	$dominant_color = wp_get_image_editor( $file )->get_dominant_color( $default_color );
```

Comment:
The `wp_get_image_editor` function may return an error object which obviously doesn't have the `get_dominant_color` method. We need to make sure that the result of the `wp_get_image_editor` function call is an object that has the `get_dominant_color` method (which may not be true if another filter with higher priority overrides image editors) before calling it.

Reply:
We are overloading the available class so they do have the needed functions

Reply:
@pbearne What if a site has another plugin that overrides Image Editor as well? Let's say a plugin that also makes some modifications to images... something like watermarks...

Reply:
OK, Lets look for feedback from others on this

Reply:
I have added checks for the methods called